logic/delegate.js

```diff
@@ -183,6 +184,7 @@ Delegate.prototype.undo = function (trs, block, sender, cb) {
 		vote: 0
 	};
 
+	// Not sure if we need this check here; sender.nameexist is never updated anywhere.
 	if (!sender.nameexist && trs.asset.delegate.username) {
```

Comment:
@SargeKhan I opened an issue to remove these related columns: https://github.com/LiskHQ/lisk/issues/800

Reply:
@karmacoma, what do you think about the whole condition? Because by the time transaction reaches apply, undo, applyUnconfirmed and undoUnconfirmed functions, we have already done schema validations on trs.asset.delegate.username. So, having this conditon here is redundant, and should be removed. Should I create an issue for it?

Reply:
I'd say remove condition(s) in this PR then. `nameexist` can be removed in #800.